library/linux/src/json_method_codec.cc

```diff
@@ -58,6 +58,10 @@ std::unique_ptr<MethodCall> JsonMethodCodec::DecodeMethodCallInternal(
 std::unique_ptr<std::vector<uint8_t>> JsonMethodCodec::EncodeMethodCallInternal(
     const MethodCall &method_call) const {
   Json::Value message(Json::objectValue);
+  if (method_call.method_name().empty()) {
```

Comment:
Why is this code necessary now? Is this unrelated cleanup?

Reply:
This change shouldn't be made to MethodCodec.

The issue here is that `flutter/keyevent` is a MessageChannel, not a MethodChannel, on the Dart side. MethodChannel is essentially a layer over MessageChannel that establishes the name/argument structure. Making *MethodCodec allow non-Method-structured messages goes against that distinction.

The right fix, which would align with the overall work of #102 is to split out JsonMessageCodec from JsonMethodCodec, but that will also require building out support for Message-based plugins. The next step of plugin refactoring that I have planned would help lay the foundation for that.

Is support for raw key events on Linux time critical, or is this something we can revisit once I'm further along with #102? If the former, I think the best way forward in the short term to support the direction of #102 would be:

- Split MessageCodec out from MethodCodec
- Add a new class above Plugin (this would be temporary, since Plugin will be significantly changed or even eliminated as part of #102 eventually anyway) that operates on generic data messages.
- Make Plugin do the conversion from data to method call, then forward to the existing Plugin methods.
- Change PluginHandler to take in that new superclass instead.

Reply:
We can probably wait. We were trying to get the overall raw key event feature to work in a short term time frame, but we can wait on the Linux implementation side.

Reply:
Okay, let's let this PR wait for a bit then. I'm hoping to do more work on plugin refactoring next week, and then we can re-evaluate.

If someone wants to do the MessageCodec/MethodCodec split in the mean time that would be useful parallel work that we'll need regardless.

Reply:
Waiting SGTM.